editor/modes/visual-editor/block-html.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * WordPress Dependencies
+ */
+import { isEqual, omit, keys } from 'lodash';
+import { Component } from '@wordpress/element';
+import { getBlockContent, getSourcedAttributes, getBlockType, isValidBlock } from '@wordpress/blocks';
+
+/**
+ * External Dependencies
+ */
+import { connect } from 'react-redux';
+import TextareaAutosize from 'react-autosize-textarea';
+
+/**
+ * Internal Dependencies
+ */
+import { updateBlock } from '../../actions';
+import { getBlock } from '../../selectors';
+
+class BlockHTML extends Component {
+	constructor( props ) {
+		super( ...arguments );
+		this.onChange = this.onChange.bind( this );
+		this.onBlur = this.onBlur.bind( this );
+		this.state = {
+			html: props.block.isValid ? getBlockContent( props.block ) : props.block.originalContent,
+		};
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( nextProps.block.attributes, this.props.block.attributes ) ) {
+			this.setState( {
+				html: getBlockContent( nextProps.block ),
+			} );
+		}
+	}
+
+	onBlur() {
+		const blockType = getBlockType( this.props.block.name );
+		const sourcedAttributes = getSourcedAttributes( this.state.html, blockType.attributes );
+		const attributes = {
+			...( omit( this.props.block.attributes, keys( sourcedAttributes ) ) ),
```

Comment:
Why do we need `omit` here if we proceed to assign in `sourcedAttributes`? In other words, the keys of `sourcedAttributes` will already override those from `this.props.block.attributes`, so the omit is unnecessary.

Reply:
I was mistakenly thinking `Object.assign({ a: 'value' }, { a: undefined });` would return `{ a: 'value' }` but I was wrong.